### PR TITLE
sizedtype accessors

### PR DESCRIPTION
**Pull request by fbs**
This adds and uses accessors for the `is_signed` and `type` fields of `sizedtype` as part of the type system improvement.

These make it easier to improve the type system, e.g. we could add `assert(IsIntTy()` to the `IsSigned` method to ensure that sign checking is only done on integers, without having to modify the code in 1000 places (like happens now)

This is largely an automatic refactor.

### CODE REVIEW

**Comment by fbs** [comment]
Link to #878

**Comment by fbs** [comment]
@danobi  can you take a look at this one when you have time? It's size makes this one annoying

File: src/types.h [hunk 1]
```diff
@@ -102,6 +100,14 @@ class SizedType
...
   bool operator==(const SizedType &t) const;
   bool operator!=(const SizedType &t) const;
+
+  bool IsSigned(void) const;
+
+  friend std::ostream &operator<<(std::ostream &, const SizedType &);
+  friend std::ostream &operator<<(std::ostream &, Type);
+
+private:
+  bool is_signed = false;
```
Comment (danobi) [inline src/types.h]: Style nit
```suggestion
  bool is_signed_ = false;
```

Comment (fbs) [reply src/types.h]: I'll sneak this in with the follow up :)

**Pull request merged**